test/__mocks__/session.mock.ts

```diff
@@ -1494,6 +1494,7 @@ export const APPLICATION_DATA_MOCK: ApplicationData = {
 
 export const APPLICATION_DATA_MOCK_WITHOUT_UPDATE: ApplicationData = {
   ...APPLICATION_DATA_MOCK,
+  [EntityNumberKey]: undefined,
```

Comment:
waas there a reason you added this? Is there a scenario where without the key something breaks?
Because I don't think in register journey this key will be in the AppData with undefined like this

Reply:
Hmm but if without this it displays the wrong template would that not mean wrong template would be displayed in register journey? During register journey `EntityNumberKey` will not be in AppData.

Reply:
Have you tested register journey?

Reply:
yes tested the Reg journey , EntityNumberKey maps to entity_number and it will always be present as its defined in ApplicationData in /model/application.model as we share the same model in Registration and Update

Reply:
entity_number key is optional and doesn't get set or exist in ApplicationData during a register journey so I would make your test data here reflect that and not set entity_number = undefined. Then fix whatever potential issue there was around that.

Reply:
Isn't that the same as setting to undefined though? Depends on what you think is clearer.

Reply:
Not exactly from an object point of view, Object.keys would contain `EntityNumberKey` if undefined. It wouldn't if we don't define/add the key at all. 
Which was just all my original comment was, adding `[EntityNumberKey]: undefined` to a mock shouldn't be necessary here and should just use a mock without EntityNumberKey.

Reply:
Which looks like @jkayyala has done now by adding APPLICATION_DATA_REGISTRATION_MOCK. Do you still need to add this line to APPLICATION_DATA_MOCK_WITHOUT_UPDATE then @jkayyala or can it be removed?

Reply:
@corybudd-ch , added it as it was clear both(undefined and not having the key at all) works in the same way and hence added the test , but it can be removed if needed